File: test_files/struct_test.c
// Simple struct test file
#include <stdio.h>

// Basic struct definition
struct Point {
    int x;
    int y;
};

// Struct with different data types
struct Person {
    char name[50];
    int age;
    float height;
    double salary;
};

// Nested struct
struct Address {
    char street[100];
    char city[50];
    int zip_code;
};

struct Employee {
    struct Person info;
    struct Address address;
    int employee_id;
};

// Function that uses structs
struct Point create_point(int x, int y) {
    struct Point p;
    p.x = x;
    p.y = y;
    return p;
}

void print_point(struct Point p) {
    printf("Point: (%d, %d)\n", p.x, p.y);
}

int main() {
    struct Point origin = {0, 0};
    struct Point p1 = create_point(10, 20);
    
    print_point(origin);
    print_point(p1);
    
    return 0;
}

File: test_files/struct_test.rs
// Basic struct definition
#[derive(Debug, Clone, Copy)]
pub struct Point {
    pub x: i32,
    pub y: i32,
}

// Struct with different data types
#[derive(Debug, Clone)]
pub struct Person {
    pub name: [u8; 50],
    pub age: i32,
    pub height: f32,
    pub salary: f64,
}

// Nested struct
#[derive(Debug, Clone)]
pub struct Address {
    pub street: [u8; 100],
    pub city: [u8; 50],
    pub zip_code: i32,
}

#[derive(Debug, Clone)]
pub struct Employee {
    pub info: Person,
    pub address: Address,
    pub employee_id: i32,
}

// Function that uses structs
pub fn create_point(x: i32, y: i32) -> Point {
    Point { x, y }
}

pub fn print_point(p: Point) {
    println!("Point: ({}, {})", p.x, p.y);
}

fn main() {
    let origin = Point { x: 0, y: 0 };
    let p1 = create_point(10, 20);

    print_point(origin);
    print_point(p1);
}
